### PR TITLE
Add isModeEnabledFn to zoom options

### PR DESCRIPTION
**Pull request by davidswinegar**
Add an optional function property, isModeEnabledFn to the zoom
options. This allows disabling modes based on other keyboard
properties.
Ex. disabling the 'y' direction when a particular key is pressed

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -68,6 +68,13 @@ plugins: {
 			// Eg. 'y' would only allow zooming in the y direction
 			mode: 'xy',
 
+			// Optional function for disabling the enabled mode, given an event.
+			// Allows disabling modes based on other keyboard properties.
+			// Ex. disabling the 'y' direction when a particular key is pressed.
+			// event: nullable DOM event
+			// direction: 'x' or 'y'
+			isModeEnabledFn: (event, direction) => true,
```
Comment (vidaaudrey) [inline README.md]: I wonder if there is a simpler approach as it's a little complicate to understand the logic, e.g. if `mode` is `xy`, and `isModeEnabledFn` returns false, which mode should `mode` be?

what about changing `mode` itself to be either the mode string or a function that returns the mode string?

**Review by rdgifford — APPROVED** [review]
To @vidaaudrey's point, you could do a map of isModeEnabledFns to modes. Then you know that for each mode you have one code path. Honestly though, this seems fine to me. The only modes are x and y, right?

File: README.md (hunk 1 above)
Comment (davidswinegar) [reply README.md]: I thought about that but I also wanted to rewrite as little of this library as possible :) The `isModeEnabledFn` overrides the `mode`, but only for disabling a particular mode

Comment (davidswinegar) [reply README.md]: @vidaaudrey I'm going to rename this to `isModeEnabledOverrideFn` for clarity that it's an override

Comment (vidaaudrey) [reply README.md]: We still want to merge this into the official package. Would be good to create a PR against `chartjs/chartjs-plugin-zoom` and get it merged there. The maintainer might have some additional feedbacks.

**Pull request merged**